buildSrc/src/main/java/org/elasticsearch/gradle/precommit/ValidateRestSpecPlugin.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.gradle.precommit;
+
+import org.elasticsearch.gradle.util.Util;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.file.FileTree;
+import org.gradle.api.provider.Provider;
+
+import java.io.File;
+
+public class ValidateRestSpecPlugin implements Plugin<Project> {
+    private static final String SCHEMA_PROJECT = ":rest-api-spec";
+    private static final String DOUBLE_STAR = "**"; // checkstyle thinks these are javadocs :(
+    // no need to find this file multiple times
+    private static volatile File jsonSchema;
+
+    @Override
+    public void apply(Project project) {
+        if (jsonSchema == null) {
```

Comment:
Firstly, project configuration is never done in parallel, so there's no need for the thread safety stuff here. Second, I think this is overly complex. Instead of all this nastiness to try and _find_ the file, let's just go ahead and have the project configure its location. It's one file, and it's rarely if every going to move. Doing a bunch of work at configuraiton time to _scan_ for this file seems unnecessary.

Reply:
I moved the work into the task configuration ... is that what you meant ?

Reply:
Not exactly, although this is better since we now defer the work unless the task is actually executed. I mean we should remove this "search" altogether and just do something along the lines of `validateJson.jsonSchema = file('rest-api-spec/src/main/resources/schema.json')` in the project build script or even hard-code in the plugin for now (since it's already tightly coupled to our project).

My reasoning being, there is exactly one file, we always know where it'll be and it rarely moves. Why all the work to dynamically locate it?

Reply:
OK .. i understand now. I hard coded it in the plugin. https://github.com/elastic/elasticsearch/pull/55117/commits/1768c6d21bd38ec6d3064b5d33ed69937c97fde2